core/finality-grandpa/src/until_imported.rs

```diff
@@ -64,23 +70,31 @@ pub(crate) trait BlockUntilImported<Block: BlockT>: Sized {
 }
 
 /// Buffering imported messages until blocks with given hashes are imported.
-pub(crate) struct UntilImported<Block: BlockT, Status, I, M: BlockUntilImported<Block>> {
+pub(crate) struct UntilImported<Block: BlockT, BlockStatus, BlockSyncRequester, I, M: BlockUntilImported<Block>> {
 	import_notifications: Fuse<Box<dyn Stream<Item = BlockImportNotification<Block>, Error = ()> + Send>>,
-	status_check: Status,
+	block_sync_requester: BlockSyncRequester,
+	status_check: BlockStatus,
+	// TODO: Why is this called inner? Why not being more descriptive and say finality_msg_stream?
```

Comment:
This is used with both the global per-set streams (commits, catch-ups) and the round streams (votes). Maybe rename it to `msg_stream`?

Reply:
I removed the TODO but we can address this in a follow up PR if this one gets merged soon.